code/edge/src/main/java/com/adobe/marketing/mobile/EdgeExtension.java

```diff
@@ -609,32 +404,18 @@ public void error(ExtensionError extensionError) {
 	 * @return the Identity shared state or null if it is pending
 	 */
 	private Map<String, Object> getIdentity(final Event event) {
-		Map<String, Object> identityXDMSharedState = getApi()
-			.getXDMSharedEventState(
-				EdgeConstants.SharedState.IDENTITY,
-				event,
-				new ExtensionErrorCallback<ExtensionError>() {
-					@Override
-					public void error(ExtensionError extensionError) {
-						MobileCore.log(
-							LoggingMode.WARNING,
-							LOG_TAG,
-							"EdgeExtension - Failed to retrieve Identity shared state with error " +
-							extensionError.getErrorName()
-						);
-					}
-				}
-			);
-
-		if (identityXDMSharedState == null) {
-			MobileCore.log(
-				LoggingMode.DEBUG,
+		SharedStateResult sharedStateResult = getApi()
+			.getXDMSharedState(EdgeConstants.SharedState.IDENTITY, event, false, SharedStateResolution.ANY);
+		if (sharedStateResult == null || sharedStateResult.getStatus() != SharedStateStatus.SET) {
```

Comment:
do we need to check the status again since it was verified in readyForEvent or can we go ahead and fetch the value directly

Reply:
It's safer to check and I don't think it harms performance in any meaningful way. In iOS we nil check the shared state as well, it's just that the Swift language semantics makes it a one liner.